ext/spl/spl_directory.c

```diff
@@ -1887,7 +1887,7 @@ static zend_result spl_filesystem_object_cast(zend_object *readobj, zval *writeo
 }
 /* }}} */
 
-static zend_result spl_filesystem_file_read_ex(spl_filesystem_object *intern, bool silent, zend_long line_add) /* {{{ */
+static zend_result spl_filesystem_file_read_ex(spl_filesystem_object *intern, bool silent, zend_long line_add, bool csv) /* {{{ */
```

Comment:
One thing I was thinking was instead of a csv bool arg, to pass the SPL flags as an argument and ``spl_filesystem_file_read_csv()`` to add the ``SPL_FILE_OBJECT_READ_CSV`` flag when calling this function. Not sure if this is cleaner tho

Reply:
Hmm, I guess we could even temporarily add the `SPL_FILE_OBJECT_READ_CSV` flag. But I'm afraid it's somewhat messy either way.